### ISSUE TITLE
Problem with Auto fill data 

### ISSUE THREAD

**Issue by ninjoan** [mautic/mautic#4047]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Y
| Feature request? | N
| Enhancement? | N

## Description:
The more details the better...

## If a bug:

| Q   | A
| --- | ---
| Mautic version |2.7.1 
| PHP version | 7.0

### Steps to reproduce:
I have a segment in my Mautic with contacts i have the basic information about this contact (first name, last name, company, email)
1. Create a standalone form
2. Create a field sample First Name and select Auto fill data > Yes, Show when value exist > No
3. Create Form action send results to User 
And the results that are sent in the email look like this

```
First Name: {formfield=firstname}
Last Name: {formfield=lastname}
Email: lead@email.com
Phone: 555-555-5555
```
 
### Log errors: 

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._

**Comment by escopecz** [comment]
Could you please double-check that the form field name in the curly brackets are correct? Aren't they should be with underscore like `{formfield=first_name}`?

**Comment by ninjoan** [comment]
Re-tested
Using Auto fill data > Yes, Show when value exist > No
The results email look like this
```
First Name: 
Last Name: 
Email: lead@email.com
Phone: 555-555-5555
```

But if i use Auto fill data > Yes, Show when value exist > Yes ( in the First Name field)
```
First Name: Peter
Last Name: 
Email: lead@email.com
Phone: 
```
Wrapping out the Autofill date is not working his full potential since the Mautic i you have a smart form that uses previous data stored the results email only arrive with the new field that collects new info but the old info stored in Mautic is not display.

**Comment by escopecz** [comment]
The auto fill configuration doesn't have any effect on the "Send results email" action. It's configuration when the form is displayed to a contact. There is another PR to use `{contactfield=...}` tokens in the results email. Would it solve your issue?

**Comment by ninjoan** [comment]
Update using `{contactfield=email}` or other fields don't work.
```
First Name: 
Last Name: 
Email: 
Phone: 
```

**Comment by escopecz** [comment]
Yes, contactfield tokens don't work. I was saying that there is another PR requesting this and I was asking if it would solve your issue. Would it?

**Comment by ninjoan** [comment]
You mean this PR #3041 ?
No that PR not solved the issue.

**Comment by escopecz** [comment]
Ah, sorry. Not PR, but issue. Yes, that one. So adding the  {contactfield=...} tokens would not your issue? What's your suggested solution?

**Comment by ninjoan** [comment]
Right now i dont think this is a bug so i think this would be more a Featured.

The idea is if you have a good contact base and you are running some campaign to update or get some field from you concurrent contact base to receive the email with the full contact information the new and the old information in Mautic


First Name: Peter (old information already in the system)
Last Name:  Pan (old information already in the system)
Email: lead@email.com (old information already in the system)
Phone: **1-555-5555** (new information)
Company:  **Acme,Inc** (new information)

**Comment by escopecz** [comment]
I can see this can be done by allowing `{contactfield=...}` tokens in that email. But you say it's not the solution. So what is?

**Comment by ninjoan** [comment]
if i put those tokens don't display nothing.

**Comment by escopecz** [comment]
Yes! It is not implemented yet! I am asking if when those tokens would work, would it solve the problem for you?
